dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/domain/Serializer.kt

```diff
@@ -0,0 +1,15 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2019 Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.domain
+
+/**
+ * The Serializer<T> generic interface. Should be implemented by any custom serializer.
+ */
+internal interface Serializer<T : Any> {
+
+    fun serialize(model: T): String
```

Comment:
This signature is a bit misleading - it says `String`, but we mean `JSON String`. Maybe the interface should be named `JSONSerializer`?